### ISSUE TITLE
WW large scenery Big Ben overrides UCES small scenery Big Ben

### ISSUE THREAD

**Issue by rctdude2** [OpenRCT2/OpenRCT2#16041]
### Operating System

Windows 10, 64 bit

### OpenRCT2 build

15ee72a

### Describe the issue

In UCES, the Beneath the Christmas Tree scenario has a single tile Big Ben object. In vanilla, it displays correctly:
![image](https://user-images.githubusercontent.com/8041567/143722866-a99dc4fa-449f-4203-973a-0feebdbf9626.png)

However in Open, it is overridden by the Wacky Worlds multi-tile Big Ben (the object _appears_ animated, presumably trying to either show all images at once or rotating between images):
![image](https://user-images.githubusercontent.com/8041567/143722891-9db42729-db84-4060-af30-35527ece8090.png)

The object file is BIGBEO.DAT which has a DAT ID of `BIGBEN  `, which is identical to the WW object.
WW BIGBEN.DAT
![image](https://user-images.githubusercontent.com/8041567/143723023-a35e384e-7d1b-4325-884d-07d774a98aab.png)
UCES BIGBEO.DAT
![image](https://user-images.githubusercontent.com/8041567/143723045-3d9dae73-efcd-45a9-bfb2-a1f70d3fc3bb.png)

```
Object conflict: 'installdir\OpenRCT2\data\object\rct2ww\scenery_large\rct2ww.scenery_large.bigben.json'
               : 'userdir\OpenRCT2\object\BIGBEO.DAT'
```

### Steps to reproduce

1. Open Beneath the Christmas Tree
2. Observe object (on top of raised tile, near back of park)

**Comment by Gymnasiast** [comment]
Most likely caused by the fact that RCT2 differed objects by both DAT name and checksum. We generally only bother with the former, but that’s obviously not working out well here.

**Comment by duncanspumpkin** [comment]
Checksums are not checked for official objects only for custom objects. I don't see any easy way of fixing this mistake made by the custom object creator.

**Comment by Gymnasiast** [comment]
It is not a mistake by the custom object creator. Even if one of the objects is official, RCT2 _does_ check the flags, which include the object type and origin - and so RCT2 can properly distinguish the two.

And even if it didn’t, I would still argue that if an object works correctly in vanilla, then it should also work in OpenRCT2.

(And a side note: the custom object was most likely created _before_ Wacky Worlds came out, as it was included in UCES which came out in 2003.)

**Comment by Gymnasiast** [comment]
> Yes that's exactly what i said. OpenRCT2 checks flags, name, (and checksum for custom objects).

We don’t - not always. In a lot of places, OpenRCT2 only looks at the DAT name and nothing else. Otherwise, this issue would not have occurred.

**Comment by duncanspumpkin** [comment]
What functions don't use the comparison function? https://github.com/OpenRCT2/OpenRCT2/blob/8d48a8279635c162613147bc9f500131892db165/src/openrct2/object/Object.cpp#L233

**Comment by rctdude2** [comment]
This object now doesn't appear at all as of at least e13a2d7, and exiting the _game_ throws a "40 images were not freed" assertion.

![image](https://user-images.githubusercontent.com/8041567/202587891-8a0c44b3-80ca-45c4-8793-f2ddba054703.png)

**Comment by duncanspumpkin** [comment]
Yes the issue is caused by a couple mistakes in the object code not storing the complete name. It's not easy to fix.

**Comment by ZeeMaji** [comment]
Was requested by gymnasiast to attach some test files to this issue.

The following park & objects are also affected by this issue.

In vanilla, these legacy path objects all show as separate entries.
<img width="1016" height="781" alt="Image" src="https://github.com/user-attachments/assets/d43f0d60-9e39-4a85-8231-780076545aa4" />

In OpenRCT2, only the first one registers, and all the others are ignored. Which results in weird effects like the same path object being listed multiple times under the paths list.
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/6c49149e-62b4-4bd6-81b3-e3cdfe847263" />

[Unnamed par2k.SC6.tar.gz](https://github.com/user-attachments/files/21844262/Unnamed.par2k.SC6.tar.gz)
[RCT1PATHS.tar.gz](https://github.com/user-attachments/files/21844269/RCT1PATHS.tar.gz)
